File: nhglib-core/src/io/github/movementspeed/nhglib/input/models/MouseSourceType.java
package io.github.movementspeed.nhglib.input.models;

import com.badlogic.gdx.Input;

/**
 * Created by Fausto Napoli on 01/03/2017.
 */
public enum MouseSourceType {
    MOUSE_XY(-1),
    MOUSE_LEFT(Input.Buttons.LEFT),
    MOUSE_RIGHT(Input.Buttons.RIGHT),
    MOUSE_MIDDLE(Input.Buttons.MIDDLE);

    public int button;

    MouseSourceType(int button) {
        this.button = button;
    }

    public static MouseSourceType fromString(String value) {
        MouseSourceType res = null;

        switch (value) {
            case "mouseXY":
                res = MOUSE_XY;
                break;

            case "mouseLeft":
                res = MOUSE_LEFT;
                break;

            case "mouseRight":
                res = MOUSE_RIGHT;
                break;

            case "mouseMiddle":
                res = MOUSE_MIDDLE;
                break;
        }

        return res;
    }

    public static MouseSourceType fromButtonCode(int button) {
        MouseSourceType res = null;

        for (MouseSourceType mouseSourceType : MouseSourceType.values()) {
            if (mouseSourceType.button == button) {
                res = mouseSourceType;
                break;
            }
        }

        return res;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/enums/MouseSourceType.kt
package io.github.movementspeed.nhglib.input.enums

import com.badlogic.gdx.Input

/**
 * Created by Fausto Napoli on 01/03/2017.
 */
enum class MouseSourceType(var button: Int) {
    MOUSE_XY(-1),
    MOUSE_LEFT(Input.Buttons.LEFT),
    MOUSE_RIGHT(Input.Buttons.RIGHT),
    MOUSE_MIDDLE(Input.Buttons.MIDDLE);

    companion object {
        fun fromString(value: String): MouseSourceType? {
            var res: MouseSourceType? = null

            when (value) {
                "mouseXY" -> res = MOUSE_XY
                "mouseLeft" -> res = MOUSE_LEFT
                "mouseRight" -> res = MOUSE_RIGHT
                "mouseMiddle" -> res = MOUSE_MIDDLE
            }

            return res
        }

        fun fromButtonCode(button: Int): MouseSourceType? {
            var res: MouseSourceType? = null

            for (mouseSourceType in values()) {
                if (mouseSourceType.button == button) {
                    res = mouseSourceType
                    break
                }
            }

            return res
        }
    }
}
